scripts/src/javatests/org/oppia/android/scripts/docs/KDocValidityCheckTest.kt

```diff
@@ -0,0 +1,796 @@
+  }
+
+  @Test
+  fun testKDoc_constructorMissingKDoc_checkShouldFail() {
+    val testContent =
+      """
+      /** Test KDoc. */
+      class TestClass {
+        constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
+      }
+      """.trimIndent()
+    val tempFile = tempFolder.newFile("testfiles/TempFile.kt")
+    tempFile.writeText(testContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(KDOC_CHECK_FAILED_OUTPUT_INDICATOR)
+    val failureMessage =
+      """
+      KDoc missing for files:
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.kt:3
+      """.trimIndent()
+    assertThat(outContent.toString().trim()).isEqualTo(failureMessage)
+  }
+
+  @Test
+  fun testKDoc_enumMissingKDoc_checkShouldFail() {
+    val testContent =
+      """
+      enum class WalkthroughPages(val value: Int) {
+        WELCOME(0),
+        TOPIC_LIST(1),
+        FINAL(2)
+      }  
+      """.trimIndent()
+    val tempFile = tempFolder.newFile("testfiles/TempFile.kt")
+    tempFile.writeText(testContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(KDOC_CHECK_FAILED_OUTPUT_INDICATOR)
+    val failureMessage =
+      """
+      KDoc missing for files:
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.kt:1
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.kt:2
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.kt:3
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.kt:4
+      """.trimIndent()
+    assertThat(outContent.toString().trim()).isEqualTo(failureMessage)
+  }
+
+  @Test
+  fun testKDoc_nestedInterfaceMissingKDoc_checkShouldFail() {
+    val testContent =
+      """
+      interface ChapterSelector {
+        
+        fun chapterSelected(chapterIndex: Int, nextStoryIndex: Int, explorationId: String)
+      
+        fun chapterUnselected(chapterIndex: Int, nextStoryIndex: Int)
+        
+        interface ChildInterface {
+          fun testFunction()
+        }
+      }  
+      """.trimIndent()
+    val tempFile = tempFolder.newFile("testfiles/TempFile.kt")
+    tempFile.writeText(testContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(KDOC_CHECK_FAILED_OUTPUT_INDICATOR)
+    val failureMessage =
+      """
+      KDoc missing for files:
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.kt:1
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.kt:3
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.kt:5
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.kt:7
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.kt:8
+      """.trimIndent()
+    assertThat(outContent.toString().trim()).isEqualTo(failureMessage)
+  }
+
+  @Test
+  fun testKDoc_companionObjectMissingKDoc_checkShouldFail() {
```

Comment:
I suggest making this test specific to just the companion object and add another test for a companion object with constituent values & functions to make it more clear the specific expected behavior for companion objects.

Reply:
Done.